app/uk/gov/hmrc/apiplatformevents/services/ApplicationEventsService.scala

```diff
@@ -18,26 +18,26 @@ package uk.gov.hmrc.apiplatformevents.services
 
 import com.google.inject.Singleton
 import javax.inject.Inject
-import uk.gov.hmrc.apiplatformevents.models.ApiPlatformEventsModel
-import uk.gov.hmrc.apiplatformevents.repository.{
-  ApiPlatformEventsDBModel,
-  ApiPlatformEventsRepository
-}
+import play.api.Logger
+import uk.gov.hmrc.apiplatformevents.models.TeamMemberAddedEvent
+import uk.gov.hmrc.apiplatformevents.repository.ApplicationEventsRepository
 import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
 @Singleton
-class ApiPlatformEventService @Inject()(repo: ApiPlatformEventsRepository) {
+class ApplicationEventsService @Inject()(repo: ApplicationEventsRepository) {
 
-  private[services] def captureEvent(event: ApiPlatformEventsModel)(
+  def captureEvent(event: TeamMemberAddedEvent)(
       implicit hc: HeaderCarrier,
-      ec: ExecutionContext): Future[Unit] = {
-    repo.createEntity(
-      ApiPlatformEventsDBModel(event.parameter1,
-                               event.parameter2,
-                               event.telephoneNumber,
-                               event.emailAddress)
-    )
+      ec: ExecutionContext): Future[Boolean] = {
+    repo.createEntity(event).recover {
+      case NonFatal(e) => {
+        Logger.info("Exception happened when trying to createEntity:",e)
+        false
```

Comment:
I find it easier to recover at the controller level as it's at that point that you know what kind of response you want to return (so for example, a 404 for a NotFoundException and a 500 for any other exception you wouldn't expect)

Reply:
will change